tsdb/index/postings.go

```diff
@@ -79,6 +79,70 @@ func (p *MemPostings) SortedKeys() []labels.Label {
 	return keys
 }
 
+type HeadStats struct {
+	Name  string
+	Count uint64
+}
+
+type Stats struct {
+	CardinalityMetricsStats []HeadStats
+	CardinalityLabelStats   []HeadStats
+	LabelValueStats         []HeadStats
+	LabelValuePairsStats    []HeadStats
+}
+
+// Calculating Cardinality Stats
+func (p *MemPostings) CardinalityStats(label string) *Stats {
+	metrics := make([]HeadStats, 0)
+	labels := make([]HeadStats, 0)
+	labelValueLenght := make([]HeadStats, 0)
+	labelValuePairs := make([]HeadStats, 0)
```

Comment:
These either need to be initialized more intelligently or replaced with a heap/slice of 10 entries that are continuously updated.